File: ShellingLaptopApp/app/src/main/java/com/shellinglaptop/adapter/CartAdapter.java
package com.shellinglaptop.adapter;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.shellinglaptop.R;
import com.shellinglaptop.databinding.ItemCartBinding;
import com.shellinglaptop.db.CartDao;
import com.shellinglaptop.db.CartDatabase;
import com.shellinglaptop.model.Cart;
import com.shellinglaptop.model.Order;
import com.shellinglaptop.utils.PriceUtils;

import java.util.List;

public class CartAdapter extends RecyclerView.Adapter<CartAdapter.CartViewHolder>{

    private List<Cart> carts;
    private ButtonOnClickListener buttonOnClickListener;

    public CartAdapter(List<Cart> carts, ButtonOnClickListener buttonOnClickListener) {
        this.carts = carts;
        this.buttonOnClickListener = buttonOnClickListener;
    }

    public void setCarts(List<Cart> carts) {
        this.carts = carts;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public CartViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new CartViewHolder(
                DataBindingUtil.inflate(LayoutInflater.from(parent.getContext()), R.layout.item_cart, parent, false),
                buttonOnClickListener
        );
    }

    @Override
    public void onBindViewHolder(@NonNull CartViewHolder holder, int position) {
        carts.get(position).setTotalMoneyStr(carts.get(position).getTotalMoney().toString());
        String setupTotalMoney = PriceUtils.setupPrice(carts.get(position).getTotalMoneyStr());
        carts.get(position).setTotalMoneyStr(setupTotalMoney +" VND");
        holder.itemCartBinding.setCart(carts.get(position));

        holder.itemCartBinding.btnDelete.setOnClickListener(v -> {
            buttonOnClickListener.btnDeleteOnClick(carts.get(position));
        });

        holder.itemCartBinding.btnOrder.setOnClickListener(v -> {
            buttonOnClickListener.btnOrderOnClick(carts.get(position));
        });
    }

    @Override
    public int getItemCount() {
        return carts.size();
    }

    class CartViewHolder extends RecyclerView.ViewHolder{

        private ItemCartBinding itemCartBinding;
        private ButtonOnClickListener buttonOnClickListener;

        public CartViewHolder(ItemCartBinding itemCartBinding, ButtonOnClickListener buttonOnClickListener) {
            super(itemCartBinding.getRoot());
            this.itemCartBinding = itemCartBinding;
            this.buttonOnClickListener = buttonOnClickListener;
        }
    }

    public interface ButtonOnClickListener{
        void btnDeleteOnClick(Cart cart);
        void btnOrderOnClick(Cart cart);
    }
}

File: ShellingLaptopApp/app/src/main/java/com/example/shellinglaptopapp/ui/mycart/CartAdapter.kt
package com.example.shellinglaptopapp.ui.mycart

import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.example.shellinglaptopapp.R
import com.example.shellinglaptopapp.data.model.Cart
import com.example.shellinglaptopapp.databinding.ItemCartBinding

class CartAdapter(
    private val carts: List<Cart>
): RecyclerView.Adapter<CartAdapter.CartViewHolder>() {

    class CartViewHolder(
        val itemCartBinding: ItemCartBinding
    ) : RecyclerView.ViewHolder(itemCartBinding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CartViewHolder =
        CartViewHolder(
            DataBindingUtil.inflate(
                LayoutInflater.from(parent.context),
                R.layout.item_cart,
                parent,
                false
            )
        )

    override fun onBindViewHolder(holder: CartViewHolder, position: Int) {
        var reverseTotalMoney = reverseString(carts[position].totalMoney.toString())

        var newTotalMoneyStr = "" //add "."
        var count = 0
        for(element in reverseTotalMoney){

            if(count == 3){
                newTotalMoneyStr = newTotalMoneyStr.plus(".")
                count = 0
            }
            count++
            newTotalMoneyStr = newTotalMoneyStr.plus(element)
        }

        var reverseNewTotalMoneyStr = reverseString(newTotalMoneyStr)

        carts[position].totalMoneyStr = "$reverseNewTotalMoneyStr VNĐ"
        holder.itemCartBinding.cart = carts[position]
    }

    private fun reverseString(str: String): String{
        var position = str.length-1
        var strReverse = ""
        for(e in str){
            strReverse = strReverse.plus(str[position])
            position--
        }

        return strReverse
    }

    override fun getItemCount(): Int = carts.size
}